blog/content/post/2023-08-21-twpm-testing.md

```diff
@@ -0,0 +1,78 @@
+---
+title: How TwPM is tested
+abstract: 'This post shows how we tested simple commands on TwPM, which is our
+          attempt at making TPM more open.'
+cover: /covers/image-file.png
+author: krystian.hebel
+layout: post
+published: true
+date: 2023-08-21
+archives: "2023"
+
+tags:
+  - TPM
+  - TwPM
+  - testing
+categories:
+  - Firmware
+  - Security
+
+---
+
+TwPM project aims to increase the trustworthiness of the TPM module (hence the
+TwPM), by providing the open-source firmware implementation for the TPM device,
+compliant to the TCG PC Client Specification. Project aims to use already
+available open-source software components whenever possible (such as TPM
+simulators for TPM commands handling), while developing new code when necessary
+(such as LPC FPGA module, or low-level TPM FIFO interface handling).
+
+This project was funded through the [NGI Assure Fund](https://nlnet.nl/assure),
+a fund established by [NLnet](https://nlnet.nl/) with financial support from the
+European Commission's [Next Generation Internet](https://ngi.eu/) programme,
+under the aegis of DG Communications Networks, Content and Technology under
+grant agreement No 957073.
+
+The project just entered a phase in which all basic components can communicate
+with each other. This should be enough to test simplest commands that don't
+require use of parts that aren't implemented yet. Nonvolatile storage, true
```

Comment:
Non-volatile storage already works on EOS S3

Reply:
This no longer applies, we switched to Orange Crab.